"Found argument '--alloc' which wasn't expected" in loader build.rs

Got the following error:

```
error: failed to run custom build command for `loader v0.1.0 (C:\Users\renhe\Documents\GitHub\psx-sdk-rs\examples\loader)`

Caused by:
  process didn't exit successfully: `C:\Users\renhe\Documents\GitHub\psx-sdk-rs\examples\loader\target\release\build\loader-63330a19a9493fb1\build-script-build` (exit code: 2)
  --- stdout
  cargo:rerun-if-changed=C:\Users\renhe\Documents\GitHub\psx-sdk-rs\examples\ferris

  --- stderr
  error: Found argument '--alloc' which wasn't expected, or isn't valid in this context

        If you tried to supply `--alloc` as a value rather than a flag, use `-- --alloc`

  USAGE:
      cargo-psx.exe --lto <PSX> <build|check|run|test>
```

Seems it's because of the `--alloc` flag being removed in [this](https://github.com/ayrtonm/psx-sdk-rs/commit/7d4c75949ded65e6a085a80c76326f31afff0b86) commit.

I tried removing the flag, and instead get these errors:

```
error[E0425]: cannot find value `MAIN_RAM` in this scope
  --> src\main.rs:29:34
   |
29 |         let load_addr = 524288 + MAIN_RAM as usize + BIOS_LEN - 2048;
   |                                  ^^^^^^^^ not found in this scope

error[E0308]: mismatched types
   --> src\main.rs:39:19
    |
39  |         file.read(exe).expect("Could not read PROG2.EXE");
    |              ---- ^^^ expected `u8`, found `u32`
    |              |
    |              arguments to this method are incorrect
    |
    = note: expected mutable reference `&mut [u8]`
               found mutable reference `&mut [u32]`
note: associated function defined here
   --> C:\Users\renhe\Documents\GitHub\psx-sdk-rs\psx\src\sys\fs.rs:339:12
    |
339 |     pub fn read(&self, dst: &mut [u8]) -> Result<usize, Error<T>> {
    |            ^^^^
```

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -50,3 +50,15 @@ jobs:
             | tee bios_stdout &
         sleep 15
         grep "test result: ok. [0-9]* passed; 0 failed" bios_stdout
+        popd
+    - name: Build examples
+      run: |
+        pushd examples
+        for d in */; do
+            if [ $d != "bios/" ]; then
+                pushd $d
+                cargo psx build
+                popd
+            fi
+        done
+        popd

--- a/examples/loader/build.rs
+++ b/examples/loader/build.rs
@@ -15,9 +15,8 @@ fn main() {
             "psx",
             "build",
             "--lto",
-            "--alloc",
-            "--cargo-args",
-            "features psx/loadable_exe",
+            "--features",
+            "psx/loadable_exe,psx/heap",
             "--load-offset=524288",
             "--stack-pointer=0",
         ])

--- a/examples/loader/src/main.rs
+++ b/examples/loader/src/main.rs
@@ -1,7 +1,6 @@
 #![no_std]
 #![no_main]
 
-use core::mem::size_of;
 use psx::constants::*;
 use psx::sys::fs::{File, CDROM};
 use psx::sys::kernel::{psx_do_execute, psx_flush_cache};
@@ -23,17 +22,15 @@ fn main() {
         const EXE_SIZE: usize =
             file_size!("../../ferris/target/mipsel-sony-psx/release/ferris.exe");
 
-        // Add the executable load offset to the end of the BIOS region (i.e. MAIN_RAM +
+        // Add the executable load offset to the end of the BIOS region (i.e. KSEG0 +
         // BIOS_LEN). The load offset refers to start of the executable without the
         // header so we have to subtract 2KB to get the actual load address.
-        let load_addr = 524288 + MAIN_RAM as usize + BIOS_LEN - 2048;
+        let load_addr = 524288 + KSEG0 + BIOS_LEN - 2048;
 
         // Create a mutable reference to the memory where the executable will be loaded
         // SAFETY: No references to this memory overlap the lifetime of this slice in
         // this executable.
-        let exe = unsafe {
-            core::slice::from_raw_parts_mut(load_addr as *mut u32, EXE_SIZE / size_of::<u32>())
-        };
+        let exe = unsafe { core::slice::from_raw_parts_mut(load_addr as *mut u8, EXE_SIZE) };
 
         // Read the CD file into the memory it will run from
         file.read(exe).expect("Could not read PROG2.EXE");
@@ -42,8 +39,8 @@ fn main() {
         // pointer to the header of a valid executable.
         unsafe {
             psx_flush_cache();
-            let init_pc_offset = 4;
-            psx_do_execute(&mut exe[init_pc_offset] as *mut u32 as *mut u8, 0, 0);
+            let init_pc_offset = 16;
+            psx_do_execute(&mut exe[init_pc_offset], 0, 0);
         }
 
         // Clear whatever the demo had on the screen

